src/algorithms/algo.py

```diff
@@ -2,181 +2,204 @@
-                        label_tokenizer, original_label_to_clean_label):
-    articles = [review[MESSAGE] for review in reviews]
-    categories = lstm_classifier.predict_labels(articles, app_config, model,
-                                                article_tokenizer,
-                                                label_tokenizer)
-
-    return [
-        format_output_json(
-            review, None, None,
-            {LSTM_CATEGORY: original_label_to_clean_label[categories[index]]})
-        for index, review in enumerate(reviews)
-    ]
-
-
-def bug_feature_classification(review, app_config, topics):
-    _, category = text_match(review[MESSAGE], topics)
-    return format_output_json(review, None, None, {BUG_FEATURE: category})
-
+    # Find the category of the review
+    category_scores, category = text_match(review.message, topics)
+    # Add the category to the review's derived insight and return the review
+    review.derived_insight.category = category
+    # Add the category scores.
+    review.derived_insight.extra_properties[constants.CATEGORY_SCORES] = category_scores
+    # Return the review
+    return review
+
+
+def lstm_classification(reviews, model, article_tokenizer, label_tokenizer, cleaned_labels):
+    articles = [review.message for review in reviews]
+    # Get the categories for each of the reviews
+    categories = lstm_classifier.predict_labels(
+        articles,
+        model,
+        article_tokenizer,
+        label_tokenizer
+    )
+
+    for index, review in enumerate(reviews):
+        review.derived_insight.extra_properties[constants.LSTM_CATEGORY] = cleaned_labels[categories[index]]
+
+    return reviews
+
+
+def bug_feature_classification(review, topics):
+    _, category = text_match(review.message, topics)
+    # Add the bug-feature classification to the review's derived insight and return the review
+    review.derived_insight.extra_properties[constants.BUG_FEATURE] = category
+    # Return the review
+    return review
 
 def run_algo():
-    app_configs = open_json(
-        APP_CONFIG_FILE.format(file_name=APP_CONFIG_FILE_NAME))
-
-    # We process all algorithms on parsed data for each app
-    for app in app_configs:
-        app_config = open_json(APP_CONFIG_FILE.format(file_name=app))
-        # If the app json schema is not valid, we don't execute any thing.
-        if not utils.validate_app_config(app_config):
-            return
-        app_config = decrypt_config(app_config)
-        # Loading the REVIEW's
-        reviews = utils.open_json(
-            PARSED_INTEGRATED_REVIEW_FILE.format(app_name=app_config[APP]))
-
-        reviews = utils.filter_reviews(reviews, app_config)
+    app_configs = utils.open_json(
+        constants.APP_CONFIG_FILE.format(file_name=constants.APP_CONFIG_FILE_NAME)
+    )
+    for app_config_file in app_configs:
+        app_config = AppConfig(
+            utils.open_json(
+                app_config_file
+            )
```

Comment:
Why have we removed `validate_app_config` function

Reply:
It's not required as, we are directly converting the app_config json to a Class. If there are any issues, it will be thrown by the class itself.

But one thing that this removes is the type check as well as any range checks.